apiary.apib

```diff
@@ -406,11 +406,402 @@ MerkleDAG.
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (Error)
+        - Message: "This command can't be called directly. Try one of its subcommands."
+        - Code: 1
+
+    + Body
+
+        ```
+        {
+          "Message": "This command can't be called directly. Try one of its subcommands.",
+          "Code": 1
+        }
+        ```
+
+## get [POST /block/get{?arg}]
+
+Store data as a raw ipfs block.
+
++ Parameters
+    + arg (Multihash, required) - hash of the block to be fetched
+
++ Request No Arguments
+
+    #### curl
+
+        curl -i http://localhost:5001/api/v0/block/get
+
+    + Body
+
+        ```
+        curl -i http://localhost:5001/api/v0/block/get
+        ```
+
++ Response 400
+
+    + Headers
+
+        ```
+        Date: Fri, 29 Jan 2016 21:46:01 GMT
+        Content-Length: 26
+        Content-Type: text/plain; charset=utf-8
+        ```
+
+    + Attributes (string)
+
+    + Body
+
+        ```
+        Argument 'key' is required
+        ```
+
++ Request With Empty Argument
+
+    The response is the same if the argument is invalid. For instance:
+
+        curl -i http://localhost:5001/api/v0/block/get?arg=kitten
+
+    #### curl
+
+        curl -i http://localhost:5001/api/v0/block/get?arg=
+
+    + Body
+
+        ```
+        curl -i http://localhost:5001/api/v0/block/get?arg=
+        ```
+
++ Response 500
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Ouptut, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Ouptut, X-Chunked-Output
+        Content-Type: application/json
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Fri, 29 Jan 2016 21:46:59 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (Error)
+        - Message: "Not a valid hash"
+        - Code: 0
+
+    + Body
+
+        ```
+        {
+          "Message": "Not a valid hash",
+          "Code": 0
+        }
+        ```
+
++ Request With Valid Argument
+
+    #### curl
+
+        curl -i http://localhost:5001/api/v0/block/get?arg=QmT78zSuBmuS4z925WZfrqQ1qHaJ56DQaTfyMUF7F8ff5o
+
+    + Body
+
+        ```
+        curl -i http://localhost:5001/api/v0/block/get?arg=QmT78zSuBmuS4z925WZfrqQ1qHaJ56DQaTfyMUF7F8ff5o
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        Content-Type: text/plain
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        X-Stream-Output: 1
+        Date: Thu, 04 Feb 2016 22:51:40 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (string)
+
+    + Body
+
+        ```
+        hello world
+        ```
+
+## put [POST /block/put{?arg}]
+
+Stores input as an IPFS block
+
+'ipfs block put' is a plumbing command for storing raw IPFS blocks.
+
++ Parameters
+    + arg (string, required) - The data to be stored as an IPFS block.
+
++ Request Without Arguments
+
+    #### curl
+
+        curl -i http://localhost:5001/api/v0/block/put
```

Comment:
this is doing a get request, for post

`curl -X POST http://localhost:5001/api/v0/block/put`